contribute/developer-guide.md

```diff
@@ -0,0 +1,309 @@
+<!-- omit in toc -->
+# Developer guide
+
+This guide helps you get started developing kubi
+
+Make sure you have the following dependencies installed before setting up your developer environment:
+
+ - Git
+ - Go
+ - In this context we will use a kind cluster for the local deployment of Kubi : cluster kind {1.23 or 124}
+
+## Deploy kubi  
+
+ - Get the source code https://github.com/ca-gip/kubi
+  
+ - install go  https://go.dev/
+    
+ - Create kind cluster with version 1.24-1.26 https://kind.sigs.k8s.io/docs/user/quick-start/
+  
```

Comment:
Could we add some of this within the makefile? It looks very tedious.

Weirdly, I think this is a usage guide, not a development guide? Maybe the usage guide can stay in the README.md, while the development guide is in CONTRIBUTING.md.